plugins/module_utils/network/ios/config/service/service.py

```diff
@@ -0,0 +1,161 @@
+    ResourceModule,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    dict_merge,
+)
+
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.facts.facts import Facts
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.rm_templates.service import (
+    ServiceTemplate,
+)
+
+
+class Service(ResourceModule):
+    """
+    The ios_service class
+    """
+
+    def __init__(self, module):
+        super(Service, self).__init__(
+            empty_fact_val={},
+            facts_module=Facts(module),
+            module=module,
+            resource="service",
+            tmplt=ServiceTemplate(),
+        )
+        self.parsers = [
+            "call_home",
+            "compress_config",
+            "config",
+            "counters",
+            "dhcp",
+            "disable_ip_fast_frag",
+            "exec_callback",
+            "exec_wait",
+            "hide_telnet_addresses",
+            "internal",
+            "linenumber",
+            "log",
+            "log_hidden",
+            "nagle",
+            "old_slip_prompts",
+            "pad",
+            "pad_cmns",
+            "pad_from_xot",
+            "pad_to_xot",
+            "password_encryption",
+            "prompt",
+            "pt_vty_logging",
+            "scripting",
+            "sequence_numbers",
+            "slave_coredump",
+            "slave_log",
+            "tcp_keepalives_in",
+            "tcp_keepalives_out",
+            "telnet_zeroidle",
+            "unsupported_transceiver",
+        ]
+        self.list_parsers = [
+            "timestamps",
+        ]
+
+    def execute_module(self):
+        """Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        if self.state not in ["parsed", "gathered"]:
+            self.generate_commands()
+            self.run_commands()
+        return self.result
+
+    def generate_commands(self):
+        """Generate configuration commands to send based on
+        want, have and desired state
+        """
+        wantd = self._service_list_to_dict(self.want)
+        haved = self._service_list_to_dict(self.have)
+
+        # if state is merged, merge want onto have and then compare
+        if self.state == "merged":
+            wantd = dict_merge(haved, wantd)
+
+        # if state is deleted, empty out wantd and set haved to wantd
+        elif self.state == "deleted":
+            wantd = self._service_list_to_dict(
+                {
+                    "dhcp": True,
+                    "pad": True,
+                    "prompt": True,
+                    "slave_log": True,
+                },
+            )
+
+        # if state us replaced or overridden, force the default value in wantd
+        elif self.state == "replaced" or self.state == "overridden":
```

Comment:
Both states seems to have similar behaviour

Reply:
As previous comment: only `replaced` has been retained